lib/CXGN/BrAPI/v2/ObservationUnits.pm

```diff
@@ -635,73 +555,176 @@ sub observationunits_store {
+                type_id    => $location_type_id->cvterm_id(),
+            });
+            if ($row) {
+                print('Row value: ' . $row->value());
+                $location_id = $row->value();
+            }
+            else {
+                die {error => sprintf('Erro retrieving the location of the study'), errorCode => 500};
+            }
+
+            my $trial_design_store = CXGN::Trial::TrialDesignStore->new({
+                bcs_schema                        => $schema,
+                trial_id                          => $study_id,
+                nd_geolocation_id                 => $location_id,
+                # nd_experiment_id => $nd_experiment->nd_experiment_id(), #optional
+                is_genotyping                     => 0,
+                new_treatment_has_plant_entries   => 0,
+                new_treatment_has_subplot_entries => 0,
+                operator                          => $user_name,
+                trial_stock_type                  => 'accessions',
+                design_type                       => $design_type,
+                design                            => $study,
+                operator                          => $user_name
+            });
+
+            my $error;
+            my $validate_design_error = $trial_design_store->validate_design();
+            if ($validate_design_error) {
+                die {error => sprintf('Error validating study design: ' . $validate_design_error), errorCode => 422};
+            }
+            else {
+                $error = $trial_design_store->store();
+                if ($error) {
+                    die {error => sprintf('ERROR store: ' . $error), errorCode => 500};
+                }
+                # Refresh the trial layout property
+                my %param = ( schema => $schema, trial_id => $study_id );
+                if ($design_type eq 'genotyping_plate'){
+                    $param{experiment_type} = 'genotyping_layout';
+                } else {
+                    $param{experiment_type} = 'field_layout';
+                }
+                my $trial_layout = CXGN::Trial::TrialLayout->new(\%param);
+                $trial_layout->generate_and_cache_layout();
+            }
+        }
+    };
+
+    my $error_resp;
+    try {
+        $schema->txn_do($coderef);
+    }
+    catch {
+        print Dumper("Error: $_\n");
+        $error_resp = CXGN::BrAPI::JSONResponse->return_error($self->status, $_->{error}, $_->{errorCode} || 500);
+    };
+    if ($error_resp) { return $error_resp; }
+
+    # Refresh materialized view so data can be retrieved. This can take a while
+    $self->_refresh_matviews($dbh, $c, 5 * 60);
+
+    # Get our new OUs by name. Not ideal, but names are unique and its the quickest solution
+    my @observationUnitNames;
+    foreach my $ou (@{$data}) { push @observationUnitNames, $ou->{observationUnitName}; }
+    my $search_params = {observationUnitNames => \@observationUnitNames};
```

Comment:
shouldn't this be:
my $search_params = {observationUnitDbIds => \@observation_unit_db_ids };
since the search uses db_ids ans not names